lib/kauth/login.go

```diff
@@ -0,0 +1,121 @@
+package kauth
+
+import (
+	"context"
+	"crypto/ed25519"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"github.com/enfabrica/enkit/astore/common"
+	"github.com/enfabrica/enkit/astore/rpc/auth"
+	"github.com/enfabrica/enkit/lib/kcerts"
+	"github.com/enfabrica/enkit/lib/logger"
+	"github.com/enfabrica/enkit/lib/retry"
+	"github.com/enfabrica/enkit/lib/srand"
+	"github.com/pkg/browser"
+	"golang.org/x/crypto/nacl/box"
+	"golang.org/x/crypto/ssh"
+	"math/rand"
+)
+
+func init() {
+	browser.Stdout = nil
+	browser.Stderr = nil
+}
+
+type EnkitCredentials struct {
+	Token string
+	// The below fields can be possibly empty if the auth server does not support CA certificates.
+	CaHosts        []string
+	CAPublicKey    string
+	PrivateKey     ed25519.PrivateKey
```

Comment:
can we make this not be explicitly an ed25519.PrivateKey, but eg, the interface you created, or generic ssh object? I don't want any code outside this library to rely on the key being ed25519.